### PR TITLE
Comments api crud

### PR DESCRIPTION
**Pull request by dormi**
The rest of API actions for comments: show, create, update, destroy

**How to test:**
```
curl -u user:test http://localhost:3000/api/issues/2

curl -u user:test -X POST -H 'Content-type: application/json' -d '{"comment": {"content": "API comment"}}' http://localhost:3000/api/issues/2/comments

curl -u user:test -X PUT -H 'Content-type: application/json' -d '{"comment": {"content": "Updated!"}}' http://localhost:3000/api/issues/2/comments/25

curl -u user:test -X DELETE -H 'Content-type: application/json' http://localhost:3000/api/issues/2/comments/25
```

### CODE REVIEW

File: engines/dradis-api/app/controllers/dradis/ce/api/v1/comments_controller.rb [hunk 1]
```diff
@@ -11,27 +11,46 @@ def index
...
-          elsif params[:evidence_id]
-            Evidence.find(params[:evidence_id])
-          else
-            raise 'Polymorphic model missing!'
-          end
+        begin
+          id_param = params.keys.select { |key| key =~ /_id$/ }.first
+          id_value = params[id_param]
+
+          klass = id_param.chomp('_id').capitalize.constantize
```
Comment (aapomm) [inline engines/dradis-api/app/controllers/dradis/ce/api/v1/comments_controller.rb]: I'm a little uncertain with having the params being constantized. Is this exploitable? For example: send out a params with user_id to manipulate the User model?

Comment (dormi) [reply engines/dradis-api/app/controllers/dradis/ce/api/v1/comments_controller.rb]: You have a point.
Reverting.

File: engines/dradis-api/spec/requests/dradis/ce/api/v1/comments_spec.rb [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+        expect(response.status).to eq 401
+      end
+    end
+    describe 'GET /api/issues/:issue_id/comments/:id' do
+      it 'throws 401' do
+        get "/api/issues/#{issue.id}/comments/#{comment.id}", env: @env
+        expect(response.status).to eq 401
+      end
+    end
+    describe 'POST /api/issues/:node_id/comments' do
```
Comment (aapomm) [inline engines/dradis-api/spec/requests/dradis/ce/api/v1/comments_spec.rb]: @dormi `:issue_id`?

**Comment by etdsoft** [comment]
The way `@comment` and `@commentable` are loaded is insecure and incompatible with multi-project.

We need to go through `current_project` to ensure the user has access to the resources.

File: engines/dradis-api/app/controllers/dradis/ce/api/v1/comments_controller.rb [hunk 3]
```diff
@@ -11,16 +11,36 @@ def index
       def show; end
 
       def create
+        @comment = @commentable.comments.build(comment_params)
+        @comment.user = current_user
+        if @comment.save
+          track_created(@comment)
+          render status: 201, location: @commentable
```
Comment (etdsoft) [inline engines/dradis-api/app/controllers/dradis/ce/api/v1/comments_controller.rb]: would this work? isn't this part of the reason why you created the custom weird polymorphic helper method

**Comment by brianp** [comment]
Closing legacy pull request. PR has been open to long. If still relevant we'll reopen as an issue but create a new up to date PR.

**Pull request closed without merging**